<launch>
  <arg name="joy_type" default="DS4" doc="options: DS4, G29"/>

  <arg name="input_joy" default="/joy" />
  <arg name="input_twist" default="/localization/twist" />

  <arg name="output_control_command" default="/remote/joy/control_cmd" />
  <arg name="output_raw_control_command" default="/remote/raw_control_cmd" />
  <arg name="output_shift" default="/remote/shift_cmd" />
  <arg name="output_turn_signal" default="/remote/turn_signal_cmd" />
  <arg name="output_gate_mode" default="/remote/gate_mode_cmd" />
  <arg name="output_emergency_stop" default="/remote/emergency_stop" />
  <arg name="output_autoware_engage" default="/autoware/engage" />
  <arg name="output_vehicle_engage" default="/vehicle/engage" />

  <arg name="config_file" default="$(find autoware_joy_controller)/config/autoware_joy_controller.yaml" />

  <node pkg="autoware_joy_controller" type="autoware_joy_controller" name="autoware_joy_controller" output="screen">
    <remap from="~input/joy" to="$(arg input_joy)"/>
    <remap from="~input/twist" to="$(arg input_twist)"/>

    <remap from="~output/control_command" to="$(arg output_control_command)"/>
    <remap from="~output/raw_control_command" to="$(arg output_raw_control_command)"/>
    <remap from="~output/shift" to="$(arg output_shift)"/>
    <remap from="~output/turn_signal" to="$(arg output_turn_signal)"/>
    <remap from="~output/gate_mode" to="$(arg output_gate_mode)"/>
    <remap from="~output/emergency_stop" to="$(arg output_emergency_stop)"/>
    <remap from="~output/vehicle_engage" to="$(arg output_vehicle_engage)"/>
    <remap from="~output/autoware_engage" to="$(arg output_autoware_engage)"/>
    <remap from="~output/vehicle_engage" to="$(arg output_vehicle_engage)"/>

    <!-- tmp -->
    <remap from="~output/vehicle_cmd" to="/remote/vehicle_cmd"/>
    <remap from="~output/raw_vehicle_cmd" to="/remote/raw_vehicle_cmd"/>

    <remap from="~service/clear_emergency_stop" to="/control/vehicle_cmd_gate/clear_external_emergency_stop"/>

    <rosparam command="load" file="$(arg config_file)" />
    <param name="joy_type" value="$(arg joy_type)" />
  </node>

  <node pkg="joy" type="joy_node" name="joy_node" output="screen">
    <param name="autorepeat_rate" value="1" />
    <param name="deadzone" value="0.0" />
  </node>
</launch>
